src/transformers/pipelines/video_classification.py

```diff
@@ -0,0 +1,113 @@
+    video.
+
+    This video classification pipeline can currently be loaded from [`pipeline`] using the following task identifier:
+    `"video-classification"`.
+
+    See the list of available models on
+    [huggingface.co/models](https://huggingface.co/models?filter=video-classification).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        requires_backends(self, "decord")
+        self.check_model_type(MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING)
+
+        self.frame_sampling_rate = kwargs.pop("frame_sample_rate", 4)
+        self.num_frames = self.model.config.num_frames
+
+    def _sanitize_parameters(self, top_k=None):
+        postprocess_params = {}
+        if top_k is not None:
+            postprocess_params["top_k"] = top_k
+        return {}, {}, postprocess_params
+
+    def __call__(self, videos: Union[str, List[str]], **kwargs):
+        """
+        Assign labels to the video(s) passed as inputs.
+
+        Args:
+            videos (`str`, `List[str]`):
+                The pipeline handles three types of videos:
+
+                - A string containing a http link pointing to a video
+                - A string containing a local path to an video
+
+                The pipeline accepts either a single video or a batch of videos, which must then be passed as a string.
+                Videos in a batch must all be in the same format: all as http links or all as local paths.
+            top_k (`int`, *optional*, defaults to 5):
+                The number of top labels that will be returned by the pipeline. If the provided number is higher than
+                the number of labels available in the model configuration, it will default to the number of labels.
+
+        Return:
+            A dictionary or a list of dictionaries containing result. If the input is a single video, will return a
+            dictionary, if the input is a list of several videos, will return a list of dictionaries corresponding to
+            the videos.
+
+            The dictionaries contain the following keys:
+
+            - **label** (`str`) -- The label identified by the model.
+            - **score** (`int`) -- The score attributed by the model for that label.
+        """
+        return super().__call__(videos, **kwargs)
+
+    def preprocess(self, video):
+
+        if video.startswith("http://") or video.startswith("https://"):
+            video = BytesIO(requests.get(video).content)
+
+        videoreader = VideoReader(video, num_threads=1, ctx=cpu(0))
+        videoreader.seek(0)
+
+        start_idx = 0
+        end_idx = self.num_frames * self.frame_sampling_rate - 1
+        indices = np.linspace(start_idx, end_idx, num=self.num_frames)
+        indices = np.clip(indices, start_idx, end_idx).astype(np.int64)
+
+        video = videoreader.get_batch(indices).asnumpy()
+        video = list(video)
```

Comment:
Why casting to `list` ? We cannot work with a real tensor ?

Shouldn't the tensor already be in `torch` format ?

Reply:
This is library agnostic...its a list of numpy frames. The video classification feature extractor expects this as a list, not numpy array, which is why I did it this way

Reply:
Casting objects around creates many copies is usually a good way to slow things down unecessarily.
For video it's especially important to be as efficient as possible IMO. (Like copying small strings or 1-d token id tensor is usually not that impactful, but entire decoded videos much more so)

Reply:
Totally agree. Unfortunately, it seems the `feature_extractor` here is looking for a list of np arrays or PIL images. Passing the whole video array and processing it all at once would be much better, but it seems that's not how its implemented?

So this issue isn't really with this PR it's rooted in the processor I think. WDYT I should do?

Other option with current implem of `feature_extractor` is to do something like `video = [x.asnumpy() for x in videoreader.get_batch(indices)]` but I assume that's slower than the way I did it here.

Reply:
Could be left as another PR, but I think the feature extractor should be able to receive the array as-is, I don't see any valid reasons to be forced to receive list of PIL image (although it's nice to be able to receive that).

I looked at the feature extractor, and it does center cropping and some normalization only, it feels to me that we could do all that at load time of the video meaning it should be mostly a no-op (the normalize should be simple add on the tensor iiuc).